src/TextInput.tsx

```diff
@@ -6,10 +6,13 @@ import TextInputWrapper from './_TextInputWrapper'
 
 type NonPassthroughProps = {
   className?: string
+  // deprecate icon prop
   icon?: React.ComponentType<{className?: string}>
```

Comment:
I think using a JSDoc comment here will give us some nice in-editor warnings for free.
```suggestion
  /** @deprecated Use `leadingIcon` prop instead */
  icon?: React.ComponentType<{className?: string}>
```

Reply:
Nice will do that!